cmake/Modules/CocosBuildHelpers.cmake

```diff
@@ -33,14 +57,42 @@ function(cocos_mark_resources)
             MACOSX_PACKAGE_LOCATION "${opt_RESOURCEBASE}/${RES_LOC}"
             HEADER_FILE_ONLY 1
             )
+
+        if(XCODE OR VS)
+            string(REPLACE "/" "\\" ide_source_group "${opt_RESOURCEBASE}/${RES_LOC}")
+            source_group("${ide_source_group}" FILES ${RES_FILE})
+        endif()
     endforeach()
 endfunction()
 
+# mark the files in the sub dir of CMAKE_CURRENT_DIR
+function(cocos_mark_code_files cocos_target)
```

Comment:
Is this code is for creating groups in xcode and Visual Studio like i.e.: https://stackoverflow.com/a/33813154/4514853 ?
If yes I think that we should remove it. Why we needs this groups at all? Both editor can display files using current folders structure. It will make only cmake file complicated and harder to maintain. If we think that folders structure is not as good as it should be we should change folder structure instead of creating filters/groups for xcode/visual studio editor.

With cmake you can use any editor not every editor supports

Reply:
yes, this code is for creating groups in Xcode now, VS not test yet.

we can't remove it, for you will see all source files in one folder named "Source Files", not current folders structure when you remove it. that xcode project is generated by cmake, cmake not group source files using folders structure default. I suggest you do a test using this branch.

We don't need consider all editors, just consider Xcode and VS for now, another reason we only do group when generating Xcode and VS project, it have no bad effect on others IDE